test/listings.ts

```diff
@@ -57,9 +66,129 @@ describe('Listing', () => {
+    expect(listing.seekerId).to.equal(1);
+  });
+
+  it('fails to set seeker account with invalid blocks for proof', async () => {
+    const seekerAccount = accounts[1];
+    const seekerAddress = await seekerAccount.getAddress();
+
+    await utils.setSeekerOwnership(mockOracle, seekers, 1, seekerAddress);
+
+    const block = await ethers.provider.getBlockNumber();
+
+    const hash = ethers.utils.solidityKeccak256(
+      ['string', 'uint256', 'address', 'uint256'],
+      [await listings.getPrefix(), 1, owner, block],
+    );
+    const proofMessage = ethers.utils.arrayify(hash);
+    const proof = await seekerAccount.signMessage(proofMessage);
+
+    await expect(
+      listings.setSeekerAccount(seekerAddress, 1, 1000, proof),
+    ).to.be.revertedWith('Proof can not be set for a future block');
+
+    await utils.advanceBlock(200);
+
+    await expect(
+      listings.setSeekerAccount(seekerAddress, 1, 10, proof),
+    ).to.be.revertedWith('Proof is expired');
+  });
+
+  it('fails to set seeker account with invalid proof', async () => {
+    const seekerAccount = accounts[1];
+    const seekerAddress = await seekerAccount.getAddress();
+
+    await utils.setSeekerOwnership(mockOracle, seekers, 1, seekerAddress);
+
+    const block = await ethers.provider.getBlockNumber();
+
+    const hash = ethers.utils.solidityKeccak256(
+      ['string', 'uint256', 'address', 'uint256'],
+      [await listings.getPrefix(), 1, owner, block],
+    );
+    const proofMessage = ethers.utils.arrayify(hash);
+
+    // sign proof with wrong account
+    const proof = await accounts[0].signMessage(proofMessage);
+
+    await expect(
+      listings.setSeekerAccount(seekerAddress, 1, block, proof),
+    ).to.be.revertedWith('Proof must be signed by specified seeker account');
+  });
+
+  it("fails to set seeker account if ownership hasn't been verified", async () => {
+    const seekerAccount = accounts[1];
+    const seekerAddress = await seekerAccount.getAddress();
+
+    const block = await ethers.provider.getBlockNumber();
+
+    const prefix = await listings.getPrefix();
+    const accountAddress = await accounts[0].getAddress();
+    const proofMessage = `${prefix}:${1}:${accountAddress.toLowerCase()}:${block.toString()}`;
+
+    const signature = await seekerAccount.signMessage(proofMessage);
+
+    await expect(
+      listings.setSeekerAccount(seekerAddress, 1, block, signature),
+    ).to.be.revertedWith('Seeker account must own the specified seeker');
+  });
+
+  it('can revoke seeker account', async () => {
+    const seekerAccount = accounts[1];
+
+    await utils.setSeekerListing(
+      listings,
+      mockOracle,
+      seekers,
+      accounts[0],
+      accounts[1],
+      1,
+    );
+
+    await listings.connect(seekerAccount).revokeSeekerAccount(owner);
+
+    const listing = await listings.getListing(owner);
+
+    expect(listing.seekerAccount).to.equal(
+      '0x0000000000000000000000000000000000000000',
```

Comment:
I saw that we can also use `ethers.constants.AddressZero`, but it really depends on your opinion and preference.

```suggestion
      ethers.constants.AddressZero,
```

Reply:
This is better I think